code/modules/mob/living/carbon/carbon.dm

```diff
@@ -704,8 +704,10 @@
 		drop_all_held_items()
 		stop_pulling()
 		throw_alert("handcuffed", /obj/screen/alert/restrained/handcuffed, new_master = src.handcuffed)
+		add_event("handcuffed", /datum/happiness_event/handcuffed)
 	else
 		clear_alert("handcuffed")
+		add_event("handcuffed", /datum/happiness_event/handcuffed)

```

Comment:
use component inheritance instead of deleting and renewing it

Reply:
this is actually supposed to be clear_event. Also it doesnt renew. Add_event returns null if the event already exists